kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -30,6 +30,10 @@
 HEAP_APPID_PROD = "2388822444"
 HEAP_ENDPOINT = "https://heapanalytics.com/api/track"
 HEAP_HEADERS = {"Content-Type": "application/json"}
+KNOWN_CI_ENV_VAR_KEYS = [
+    "CODEBUILD_BUILD_ID",  # https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html
+    "JENKINS_URL",  # https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
+]
```

Comment:
This will need more variables. Dagster is Apache license, we can copy https://github.com/dagster-io/dagster/blob/e31e577b8841e0ff54745e206d621d9bf934a063/python_modules/dagster/dagster/_core/telemetry.py#L92-L101